plugins/woocommerce/src/Blocks/BlockPatterns.php

```diff
@@ -112,19 +130,53 @@ public function register_block_patterns() {
 		);
 
 		if ( ! file_exists( $this->patterns_path ) ) {
-			return;
+			return array();
 		}
 
 		$files = glob( $this->patterns_path . '/*.php' );
 		if ( ! $files ) {
-			return;
+			return array();
 		}
 
+		$patterns = array();
+
 		foreach ( $files as $file ) {
-			$pattern_data = get_file_data( $file, $default_headers );
+			$data           = get_file_data( $file, $default_headers );
+			$data['source'] = $file;
+			$patterns[]     = $data;
+		}
 
-			$this->pattern_registry->register_block_pattern( $file, $pattern_data, $this->dictionary );
+		$this->set_pattern_cache( $patterns );
+		return $patterns;
+	}
+
+	/**
+	 * Gets block pattern cache.
+	 *
+	 * @return array|false Returns an array of patterns if cache is found, otherwise false.
+	 */
+	private function get_pattern_cache() {
+		$pattern_data = get_site_transient( 'woocommerce_blocks_patterns' );
+
+		if ( is_array( $pattern_data ) && $pattern_data['version'] === WOOCOMMERCE_VERSION ) {
+			return $pattern_data['patterns'];
 		}
+
+		return false;
+	}
+
+	/**
+	 * Sets block pattern cache.
+	 *
+	 * @param array $patterns Block patterns data to set in cache.
+	 */
+	private function set_pattern_cache( array $patterns ) {
+		$pattern_data = array(
+			'version'  => WOOCOMMERCE_VERSION,
+			'patterns' => $patterns,
+		);
+
+		set_site_transient( 'woocommerce_blocks_patterns', $pattern_data, WEEK_IN_SECONDS );
```

Comment:
Since our patterns would ever get updated on a release, perhaps we can extend the caching to match our release cycle?

Reply:
I think the cache is likely to get invalidated more quickly by patch releases, but I don't see any harm in bumping this up a little bit.

Bumped to one month in https://github.com/woocommerce/woocommerce/pull/51802/commits/0dc260e5d32b6d85a17e85f21ba34df9de51a18b.